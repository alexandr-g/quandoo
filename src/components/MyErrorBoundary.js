/* eslint handle-callback-err: "warn" */

import React, { Component } from 'react'
import PropTypes from 'prop-types'

class MyErrorBoundary extends Component {
  constructor (props) {
    super(props)

    this.state = {
      hasError: false
    }
  }

  componentDidCatch (error, info) {
    this.setState(state => ({ ...state, hasError: true }))
    // in production send this to error reporting service - (Sentry, TrackJS)
    console.log(error, info)
  }

  render () {
    if (this.state.hasError) {
      return <div>Sorry, something went wrong.</div>
    } else {
      return this.props.children
    }
  }
}

MyErrorBoundary.propTypes = {
  children: PropTypes.element
}

export default MyErrorBoundary
