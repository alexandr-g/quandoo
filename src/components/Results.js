import React from 'react'
import PropTypes from 'prop-types'

const Results = ({ userInfo }) => {
  if (!userInfo) {
    return <div>Loading...</div>
  }

  const { name, login, location, bio } = userInfo

  return (
    <div className='container'>
      <div className='card-block'>
        <ul className='list-group'>
          <li className='list-group-item'>
            <b>Name:</b> {name}
          </li>
          <li className='list-group-item'>
            <b>Login:</b> {login}
          </li>
          <li className='list-group-item'>
            <b>Location:</b> {location}
          </li>
          <li className='list-group-item'>
            <b>Bio:</b> {bio}
          </li>
        </ul>
      </div>
    </div>
  )
}

Results.propTypes = {
  userInfo: PropTypes.object
}

export default Results
