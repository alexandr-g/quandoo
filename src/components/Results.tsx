import React from 'react'
import styled from 'styled-components'

interface Props {
  userInfo: {
    name: string,
    login: string,
    location: string,
    bio: string,
  }
}

const Results = ({ userInfo }: Props) => {
  const { name, login, location, bio } = userInfo

  return (
    <div className='container'>
      <div className='card-block'>
        <ul className='list-group'>
          <li className='list-group-item'>
            <b>Name:</b> <TextTile> {name} </TextTile>
          </li>
          <li className='list-group-item'>
            <b>Login:</b> <TextTile> {login} </TextTile>
          </li>
          <li className='list-group-item'>
            <b>Location:</b> <TextTile> {location} </TextTile>
          </li>
          <li className='list-group-item'>
            <b>Bio:</b> <TextTile> {bio} </TextTile>
          </li>
        </ul>
      </div>
    </div>
  )
}

const TextTile = styled.span`
  padding-left: 7px
`

export default Results
