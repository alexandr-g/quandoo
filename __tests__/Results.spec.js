/* global describe, test, expect */

import React from 'react'
import Results from '../src/components/Results'
import renderer from 'react-test-renderer'
import MyErrorBoundary from '../src/components/MyErrorBoundary'

describe('Results component', () => {
  test('results list is rendered without user info object', () => {
    const component = renderer.create(<MyErrorBoundary><Results /></MyErrorBoundary>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('results list is rendered with user info object', () => {
    const userInfo = {
      name: 'test',
      login: 'test login',
      location: 'planet earth'
    }
    const component = renderer.create(<Results userInfo={userInfo} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
