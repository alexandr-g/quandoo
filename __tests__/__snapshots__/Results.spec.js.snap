// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`results list is rendered with user info object 1`] = `
<div
  className="container"
>
  <div
    className="card-block"
  >
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        Name: 
        test
      </li>
      <li
        className="list-group-item"
      >
        Login: 
        test login
      </li>
      <li
        className="list-group-item"
      >
        Location: 
        planet earth
      </li>
      <li
        className="list-group-item"
      >
        Bio: 
      </li>
    </ul>
  </div>
</div>
`;

exports[`results list is rendered without user info object 1`] = `
<div>
  Loading...
</div>
`;
